client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/content/get-titan-footer-details.ts

```diff
@@ -1,16 +1,42 @@
+import { HelpCenter } from '@automattic/data-stores';
 import { localizeUrl } from '@automattic/i18n-utils';
+import { useDispatch } from '@wordpress/data';
 import { translate } from 'i18n-calypso';
 import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
 import { TITAN_CONTROL_PANEL_CONTEXT_GET_MOBILE_APP } from 'calypso/lib/titan/constants';
 import { recordEmailAppLaunchEvent } from 'calypso/my-sites/email/email-management/home/utils';
 import { getTitanControlPanelRedirectPath } from 'calypso/my-sites/email/paths';
 import type { ThankYouFooterDetailProps } from 'calypso/components/thank-you-v2/footer';
 
-export default function getTitanFooterDetails(
+const HELP_CENTER_STORE = HelpCenter.register();
+const SUPPORT_SITE_ID = 9619154;
+const TITAN_SUPPORT_PAGE_ID = 370395;
+
+export default function useGetTitanFooterDetails(
+	selectedSiteSlug: string | null,
+	domainName: string,
+	currentRoute: string,
+	context: string,
+	limit?: number
+) {
+	const { setShowSupportDoc } = useDispatch( HELP_CENTER_STORE );
+
+	return getTitanFooterDetails(
+		selectedSiteSlug,
+		domainName,
+		currentRoute,
+		context,
+		setShowSupportDoc,
+		limit
+	);
+}
```

Comment:
Interesting approach, but it's just a workaround:
1) `getTitanFooterDetails` returns simple array, not a react component or hook, but this way we just lie to React and there is bug chance to have consequences in the future due to it.
2) Also in code we use it as a simple function but it's defined as a hook, so it can confuse Code Wranglers who will work with this code in the future.
3) Also it adds a lot of extra code

To handle it in a more proper way - we can update [ThankYouFooterDatail](https://github.com/Automattic/wp-calypso/blob/trunk/client/components/thank-you-v2/footer/index.tsx#L26-L39), with the help of expecting property `supportDocHref` and running it as:

```
if ( buttonText && ( buttonHref || buttonOnClick || supportDocHref ) ) {
...
	onClick={() => {
		supportDocHref && setShowSupportDoc(supportDocHref);
		buttonOnClick?.();
	}}
...
```

Reply:
Thanks @nightnei, that does clean things up quite a bit. I had to make a few adjustments to this approach to make sure we we're including the support doc ID and not just the URL, as well as allow for cases where we're just opening a basic Help Center instance and not a specific doc.

Should be ready for review. I did have a little trouble with the non Help Center links (Titan flow, get the app as well as the Theme selection after purchasing a plan) but I think that's a local caching issue I'm experiencing, not something related to the PR. Let me know if those links give you trouble as well though!